File: clients/desktop/GraNAS.Desktop.App/Services/Api/AuthApi.cs
using GraNAS.Desktop.Contracts.Auth;

namespace GraNAS.Desktop.App.Services.Api;

public class AuthApi : ApiBase, IAuthApi
{
  public AuthApi(HttpClient http) : base(http) { }

  public async Task<TokenResponse?> LoginAsync(LoginRequest request, CancellationToken ct = default)
  {
    try { return await PostAsync<TokenResponse>("api/auth/login", request, ct); }
    catch (ApiException ex) when (ex.StatusCode == 401) { return null; }
  }

  public Task RegisterAsync(RegisterRequest request, CancellationToken ct = default)
    => PostAsync("api/auth/register", request, ct);

  public async Task<TokenResponse?> RefreshAsync(string refreshToken, CancellationToken ct = default)
  {
    try { return await PostAsync<TokenResponse>("api/auth/refresh", new { refreshToken }, ct); }
    catch (ApiException ex) when (ex.StatusCode == 401) { return null; }
  }

  public Task LogoutAsync(string? refreshToken, CancellationToken ct = default)
    => PostAsync("api/auth/logout", new { refreshToken }, ct);

  public async Task<MeResponse?> MeAsync(CancellationToken ct = default)
  {
    try { return await GetAsync<MeResponse>("api/auth/me", ct); }
    catch { return null; }
  }
}

File: clients/web/src/api/auth.api.ts
import type { LoginRequest, RegisterRequest, RegisterResponse, TokenResponse } from '../types/auth';
import { api } from './client';

export const authApi = {
  register: (data: RegisterRequest) =>
    api.post<RegisterResponse>('/api/auth/register', data),

  login: (data: LoginRequest) =>
    api.post<TokenResponse>('/api/auth/login', data),

  logout: () =>
    api.post<{ message: string }>('/api/auth/logout'),
};
